### PR TITLE
Handle signals in dist mode

### PR DESCRIPTION
**Pull request by drahnr**
Handles signals in dist mode by adding them as a future and `select!`.

Ref #1620

Orignal work by @montekki

### CODE REVIEW

**Comment by sylvestre** [comment]
should we manage the HUP signal ?

**Comment by drahnr** [comment]
We could and should, but I think that's a separate concern for another PR and requires some additional logic

**Comment by sylvestre** [comment]
makes sense
I guess you saw that it fails to build

File: src/bin/sccache-dist/main.rs [hunk 1]
```diff
@@ -67,18 +73,31 @@ fn main() {
...
+                2
             }
-            2
         }
-    });
+    };
+        
+    // Whenever `docker` or `systemd` stop the service, the signals
+    // have to be processed and the app has to perform a graceful stop.
+    select! {
```
Comment (mati865) [inline src/bin/sccache-dist/main.rs:92]: By the default select will randomly pick the order of polling. You can make it use the order you have specified (also avoiding calling random) by adding `biased`.
I'm on my mobile but it should look something like this:
```suggestion
    select! {
        biased;
        
```

**Comment by drahnr** [comment]
I'll get back to this eventually

**Pull request closed without merging**